testing/kfctl/kf_is_ready_test.py

```diff
@@ -0,0 +1,66 @@
+import datetime
+import logging
+import os
+import subprocess
+import tempfile
+import uuid
+from retrying import retry
+
+import pytest
+
+from kubeflow.testing import util
+from testing import deploy_utils
+
+def test_kf_is_ready(namespace):
+  """Test that Kubeflow was successfully deployed.
+
+  Args:
+    namespace: The namespace Kubeflow is deployed to.
+  """
+
+  logging.info("Using namespace %s", namespace)
+
+  # Need to activate account for scopes.
+  if os.getenv("GOOGLE_APPLICATION_CREDENTIALS"):
+    util.run(["gcloud", "auth", "activate-service-account",
+              "--key-file=" + os.environ["GOOGLE_APPLICATION_CREDENTIALS"]])
+
+
+
+  api_client = deploy_utils.create_k8s_client()
+
+  util.load_kube_config()
+
+  # Verify that components are actually deployed.
+  # TODO(jlewi): We need to parameterize this list based on whether
+  # we are using IAP or basic auth.
+  deployment_names = [
```

Comment:
maybe read config files from bootstrap/config/... ?

Reply:
I think it might be better to be explicit. This way its clear what the expectations are (e.g. which servicecs could be running).

If we parse the config and get a list of components dynamically then its not clear if we are setting expectations correctly. Also, if we do it that way and someone accidentally deletes some component then our E2E test wouldn't catch that.